src/charm.py

```diff
@@ -477,13 +531,59 @@ def _update_application_master(self) -> None:
         """Use Sentinel to update the current master hostname."""
         info = self.sentinel.get_master_info()
         logger.debug(f"Master info: {info}")
+        if info is None:
+            logger.warning("Could not update current master")
+            return
 
         self._peers.data[self.app][LEADER_HOST_KEY] = info["ip"]
 
+    def _sentinel_failover(self, departing_unit_name: str) -> None:
+        """Try to failover the current master."""
+        if self._k8s_hostname(departing_unit_name) != self.current_master:
+            # No failover needed
+            return
+
+        with self.sentinel.sentinel_client() as sentinel:
+            try:
+                sentinel.execute_command(f"SENTINEL FAILOVER {self._name}")
+            except RedisError:
+                raise
```

Comment:
since you're not doing anything with RedisError I am pretty sure you can just do 
```suggestion
sentinel.execute_command(f"SENTINEL FAILOVER {self._name}")
```

Reply:
Yeah I was on the fence about making it explicit or not. I'll remove it since `RedisError` is being handled on the calling method.